backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java

```diff
@@ -1003,12 +1011,49 @@ private String createInfo(@NonNull final I_SEPA_Export_Line line)
 		}
 		if (line.getC_BP_BankAccount_ID() > 0)
 		{
-			sb.append(" @C_BP_BankAccount_ID@ ");
-			sb.append(line.getC_BP_BankAccount().getDescription());
+			if (line.getC_BP_BankAccount().getDescription() != null)
+			{
+				sb.append(" @C_BP_BankAccount_ID@ ");
+				sb.append(line.getC_BP_BankAccount().getDescription());
+			}
+			if (line.getC_BP_BankAccount().getName() != null)
+			{
+				sb.append(" @Name@ ");
+				sb.append(line.getC_BP_BankAccount().getName());
+			}
+			if (line.getC_BP_BankAccount().getIBAN() != null)
+			{
+				sb.append(" @IBAN@ ");
+				sb.append(line.getC_BP_BankAccount().getIBAN());
+			}
+		}
+		if (line.getStructuredRemittanceInfo() != null)
+		{
+			sb.append(" @StructuredRemittanceInfo@ ");
+			sb.append(line.getStructuredRemittanceInfo());
 		}
+		if (line.getRecord_ID() > 0)
+		{
+			sb.append(" @C_Invoice_ID@ ");
+			sb.append(getInvoiceIDByRecordID(line));
+		}
+
 		return Services.get(IMsgBL.class).parseTranslation(InterfaceWrapperHelper.getCtx(line), sb.toString());
 	}
 
+	private final IInvoiceDAO invoiceDAO = Services.get(IInvoiceDAO.class);
+	private int getInvoiceIDByRecordID(final I_SEPA_Export_Line sepaline)
+	{
+		final TableRecordReference referencedRecord = TableRecordReference.of(
+				sepaline.getAD_Table_ID(),
+				sepaline.getRecord_ID());
+
```

Comment:
Did you test this?
Does not make any sense?
The record reference shall be the payselection line and not invoice.
And from there you shall be able to take the invoice.

Reply:
I tested that yesterday, which failes and this is why, I now have some changes that im going to push now to roll out